### PR TITLE
Add some tests to improve code coverage

### PR DESCRIPTION
**Pull request by athieriot**
Added a fair amount of tests and manage to obtain a pretty good coverage.

Obviously it did cost a little bit of refactoring but I believe it's all for the best (Mostly split into more functions, better use of Promises)

   - Introducing Gulp as build tool (This can be reverted if you prefer doing without but it's quite practical)
   - Test runner is Mocha + Istanbul to generate code coverage in LCOV format.
   - During testing, HTTP calls are intercepted and validated using [Nock](https://www.npmjs.com/package/nock) 
   - Fail the build if coverage goes bellow 90%

What I don't is how ```aircover``` will detect the LCOV file which is currently located in ```coverage/lcov.info```

Edit: Giving a try to .drone.yml configuration following conversation on drone-dev

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,6 +10,6 @@ RUN apk update && \
 WORKDIR /node
 ADD package.json /node/
 ADD index.js /node/
-RUN npm install
+RUN npm install --production
```
Comment (tboerger) [inline Dockerfile]: Please do that also for the other node based plugins.

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,12 @@
 .PHONY: install test docker
 
-IMAGE ?= plugins/drone-nuget
+IMAGE ?= plugins/drone-artifactory
 
 install:
-	npm install
+	npm install --quiet
```
Comment (tboerger) [inline Makefile]: Please do that also for the other node based plugins.

File: Dockerfile (hunk 1 above)
Comment (athieriot) [inline Dockerfile]: Will do ! (But they don't have dev dependencies at the moment so it's not as critical)

**Comment by tboerger** [comment]
LGTM

**Comment by athieriot** [comment]
LGTM

**Pull request merged**